payments-core/src/main/java/com/stripe/android/payments/bankaccount/navigation/CollectBankAccountContract.kt

```diff
@@ -104,3 +138,22 @@ class CollectBankAccountContract :
             "com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract.extra_result"
     }
 }
+
+internal fun CollectBankAccountResultInternal.toPublicType(): CollectBankAccountResult {
+    return when (this) {
+        is CollectBankAccountResultInternal.Cancelled -> {
+            CollectBankAccountResult.Cancelled
+        }
+        is CollectBankAccountResultInternal.Completed -> {
+            CollectBankAccountResult.Completed(
+                response = CollectBankAccountResponse(
+                    intent = response.intent!!,
+                    financialConnectionsSession = response.financialConnectionsSession
+                )
+            )
+        }
+        is CollectBankAccountResultInternal.Failed -> {
+            CollectBankAccountResult.Failed(error)
+        }
+    }
+}
```

Comment:
can we rename to `toExposedResult` to match the pattern used in connections? https://github.com/stripe/stripe-android/blob/73d7f24b46df6a971f65ddc79fb4e640291d8e55/financial-connections/src/main/java/com/stripe/android/financialconnections/launcher/FinancialConnectionsSheetForLinkContract.kt#L35-L34

Reply:
We can also throw a more meaningful exception rather than doing null assert (as in the method linked above)